src/lib/inet/src/linux/inet_gretap.c

```diff
@@ -208,26 +256,58 @@ bool inet_gretap_interface_start(inet_gretap_t *self, bool enable)
             return false;
         }
 
-        if (osn_ip_addr_cmp(&self->in_local_addr, &OSN_IP_ADDR_INIT) == 0)
-        {
-            LOG(INFO, "inet_gretap: %s: No local address was specified.", self->inet.in_ifname);
-            return false;
-        }
+        switch (self->family) {
+        case AF_INET:
+            if (osn_ip_addr_cmp(&self->in_local_addr, &OSN_IP_ADDR_INIT) == 0)
+            {
+                LOG(INFO, "inet_gretap: %s: No local address was specified.", self->inet.in_ifname);
+                return false;
+            }
+
+            if (osn_ip_addr_cmp(&self->in_remote_addr, &OSN_IP_ADDR_INIT) == 0)
+            {
+                LOG(INFO, "inet_gretap: %s: No remote address was specified.", self->inet.in_ifname);
+                return false;
+            }
+
+            snprintf(slocal_addr, sizeof(slocal_addr), PRI_osn_ip_addr, FMT_osn_ip_addr(self->in_local_addr));
+            snprintf(sremote_addr, sizeof(sremote_addr), PRI_osn_ip_addr, FMT_osn_ip_addr(self->in_remote_addr));
+            strncpy(family, "-4", sizeof(family));
+            strncpy(type, "gretap", sizeof(type));
```

Comment:
```suggestion
            family = "-4";
            type = "gretap";
```

strncpy() is an unsafe function (indeed it is safe here, but then again, so would be strcpy)
See: https://www.opensync.io/s/EDE-020-041-501_OpenSync_Coding_Standard.pdf

Reply:
OK